inst/tests/tests.Rraw

```diff
@@ -18059,6 +18059,13 @@ for (opt in c(0,Inf)) {
   testnum = 2233.44
 }
 options(old)
+# optimized := with gforce functions that can return lists #5403
+DT = data.table(grp=1:2, x=1:4)
+test(2233.45, DT[, `:=`(y = shift(x, type = "lag", n = 1), z = shift(x = x, type = "lead", n = 1)), by = grp], data.table(grp=1:2, x=1:4, y=c(NA,NA,1:2), z=c(3:4, NA, NA)))
```

Comment:
maybe instead of `DT = ...` repeatedly, the test could be `copy(DT)[...]`, WDYT?